### DISCUSSION TITLE
Support phase_development_build

### DISCUSSION THREAD

**Discussion by ethanwillis** [vercel/next.js#23973] [Ideas]
# Feature request
It would be nice to have a phase for a ```PHASE_DEVELOPMENT_BUILD``` where a static build can be produced where ```process.env.NODE_ENV``` is set to ```development```

## Is your feature request related to a problem? Please describe.

The reason you'd want this is so that you could do react profiling with static builds for a development environment. See [1](https://reactjs.org/docs/profiler.html), [2](https://gist.github.com/bvaughn/25e6233aeb1b4f0cdb8d8366e54a3977)

In order to enable react profiling in a static build the webpack config must be modified in the next config to alias the profiling versions of the react-dom and scheduler/tracing modules, as documented by the react team. (See ref above.)

Edit: In a more general sense if there are any third party libraries or your own code that relies on process.env.NODE_ENV being set correctly in a static build used for development this phase is needed. Otherwise it's always set to production and can't be changed because this standard environment variable has been hijacked. See: [3](https://github.com/zeit/next.js/blob/master/errors/env-key-not-allowed.md)

## Describe the solution you'd like

Ideally I'd be able to use phase information or ```options.dev``` in my next config to determine whether i want to load the profiling versions of these modules. 

## Describe alternatives you've considered

The alternative I've considered for the moment is passing in a non-standard(not NODE_ENV) environment variable to check against. However, it's possible other libraries within my project are relying on the incorrect NODE_ENV value that is set to ```production``` for all static builds.

**Comment by ethanwillis** [comment]
Related Issues
Open:
#10129 
Closed: 
#3605, #9123, #6062

**Comment by jacargentina** [comment]
To add another use case to this.

I just want/need to test locally my migrated react-app to NextJS BEFORE deploying to the real server/domain (still in production)

Currently i `have .env.development` and `.env.production` with env variables pointing to a graphql server, etc. On *development* i'm pointing to my local dev machine for everything, as i have all the testing db, api, etc, with some sample data.

Also, i must note i have a custom NextJS server, with some proxies to other site parts, so that is another thing to test before deploy.

As there is no option to use NODE_ENV=development with `next build` (so it picks up `.env.development` variables), the only workaround i have is to temporarily *overwrite* my `.env.production` file with values from `.env.development`

**Comment by JakubKoralewski** [comment]
I'd also be interested in this. My use case is staging and CI testing environments in which I'd like to have my app code respect `NODE_ENV=production` but overwrite React's StrictMode and throw error if strict mode barks while still having the guarantess of running a `NODE_ENV=production` build which will be the same as my deployment.  Also in CI I'd like to maybe explore building my TypeScript codebase in strict mode and report null/undefined errors as warnings/errors.
